### PR TITLE
Remove the `ffi-support` crate sources from this repo.

### LINKED ISSUE #4152: Make it clearer that ffi-support lives in its own repo
I see that there was a [ffi-support release](https://crates.io/crates/ffi-support/0.4.3) yesterday: v0.4.3.
It seems that release was done from the https://github.com/mozilla/ffi-support repository.

However the `repository` still points here.
Can you clarify where ffi-support's canonical repository lives now?
If its in the external repository, why does the internal code at `components/support/ffi` remains?

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-2230)
┆Sprint: Backlog

### PR DESCRIPTION
**Pull request by rfk**
As of https://github.com/mozilla/application-services/issues/3463
the `ffi-support` crate lives in its own repo, and it's confusing
having an old copy of it still hanging around in this one.

Fixes #4152.

### CODE REVIEW

File: testing/separated/ffi-support-extra-test/src/test.rs [hunk 1]
```diff
@@ -1,89 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-//! This test is a stress test meant to trigger some bugs seen prior to the use
-//! of handlemaps. See /docs/design/test-faster.md for why it's split -- TLDR:
-//! it uses rayon and is hard to rewrite with normal threads.
```
Comment (rfk) [inline testing/separated/ffi-support-extra-test/src/test.rs:7]: FYI I checked, and this test does indeed seem to have been ported over to the standalone repo, so I don't feel bad about deleting it here.

**Comment by mhammond** [comment]
I was expecting to see `components/support/ffi` be removed entirely?

**Comment by rfk** [comment]
> I was expecting to see components/support/ffi be removed entirely?

Um, yes, so was I 😅 ; perhaps I accidentally restored it while messing around with git branches...

**Comment by rfk** [comment]
@mhammond thanks, fixed!

**Review by mhammond — APPROVED** [review]

**Pull request merged**